### PR TITLE
[voq][chassis]fix recirc port admin status

### LINKED ISSUE #10761: [Voq][Chassis]  Recirc port stays admin down
#### Description
In the Voq Chassis the `Recirc` [port with role 'Rec'] stays admin-down after load minigraph.

#### Steps to reproduce the issue:
1. Load the T2 linecard minigraph on Voq chassis linecard
2. execute `sudo config load_minigraph`
3. check the admin and oper status of the recirc port 

#### Describe the results you received:
```
admin@str2-sonic-lc3-1:~$ show interface status | grep Rec0
  Ethernet-Rec0      221     400G   9100    N/A     Recirc0/0           routed    down     down              N/A         off
admin@str2-sonic-lc3-1:~$
```

#### Describe the results you expected:
The expectation is this `Recirc` port should be up after load minigraph

#### Output of `show version`:

```
SONiC Software Version: SONiC.master.90181-0191300b9
Distribution: Debian 11.3
Kernel: 5.10.0-8-2-amd64
Build commit: 0191300b9
Build date: Thu Apr 14 18:53:06 UTC 2022
Built by: AzDevOps@sonic-build-workers-001E8J
```
#### Output of `show techsupport`:

```
(paste your output here or download and attach the file here )
```

### PR DESCRIPTION
**Pull request by arlakshm**
Signed-off-by: Arvindsrinivasan Lakshmi Narasimhan <arlakshm@microsoft.com>

#### Why I did it
resolves #10761.
For VOQ chassis, the Recirc port, which was added for the Everflow, stays admin down after load minigraph.
This PR add the fix to make the recirc port as admin up
#### How I did it
The PR adds a change in minigraph.py, if port has`role` as `Rec` make the the port as admin-status up.

#### How to verify it
UT 

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

**Comment by arlakshm** [comment]
Hi @qiluo-msft , @abdosi , @judyjoseph , can please review this PR and approve if no comments

**Review by judyjoseph — APPROVED** [review]

File: src/sonic-config-engine/minigraph.py [hunk 1]
```diff
@@ -1530,6 +1530,12 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
         if inband_port in ports.keys():
             ports[inband_port]['admin_status'] = 'up'
 
+# bring up the recirc port for voq chassis
```
Comment (qiluo-msft) [inline src/sonic-config-engine/minigraph.py:1533]: ># bring [](http://example.com/codeflow?start=0&length=7)

Indent the same level as below code.

File: src/sonic-config-engine/tests/test_cfggen.py [hunk 2]
```diff
@@ -894,18 +894,43 @@ def test_minigraph_voq_inband_port(self):
...
+                'description': 'Recirc0/1',
+                'index': '52',
+                'lanes': '222',
+                'mtu': '9100',
+                'numVoq': '8',
+                'pfc_asym': 'off',
+                'role': 'Inb',
+                'speed': '400000',
+                'tpid': '0x8100'
             }
```
Comment (qiluo-msft) [inline src/sonic-config-engine/tests/test_cfggen.py:910]: Reformat the python dict, same for below function.

File: src/sonic-config-engine/tests/voq-sample-port-config.ini [hunk 3]
```diff
@@ -1,34 +1,11 @@
...
-Ethernet104     85,86,87,88       fortyGigE0/104
-Ethernet108     81,82,83,84       fortyGigE0/108
-Ethernet112     89,90,91,92       fortyGigE0/112
-Ethernet116     93,94,95,96       fortyGigE0/116
-Ethernet120     97,98,99,100      fortyGigE0/120
-Ethernet124     101,102,103,104   fortyGigE0/124
-Ethernet-IB0    133               Recycle0
+# name              lanes     alias               index  role   speed           coreId     corePortId     numVoq
+Ethernet0           6,7         Ethernet1/1           1      Ext   100000        0           1                 8
+Ethernet4           2,3          Ethernet2/1           2      Ext   100000       0           2                  8
```
Comment (qiluo-msft) [inline src/sonic-config-engine/tests/voq-sample-port-config.ini:3]: >           [](http://example.com/codeflow?start=23&length=10)

Align the columns

File: src/sonic-config-engine/minigraph.py [hunk 4]
```diff
@@ -1530,6 +1530,12 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
...
+    for port, port_attributes in ports.items():
+        port_role = port_attributes.get('role', None)
+        if port_role == 'Rec':
+            ports[port]['admin_status'] = 'up'
```
Comment (qiluo-msft) [inline src/sonic-config-engine/minigraph.py:1537]: >ports[port]['admin_status'] [](http://example.com/codeflow?start=12&length=27)

If the original value is `down`, do you intend to override?

File: src/sonic-config-engine/minigraph.py (hunk 1 above)
Comment (arlakshm) [reply src/sonic-config-engine/minigraph.py]: fixed in latest commit

File: src/sonic-config-engine/tests/test_cfggen.py (hunk 2 above)
Comment (arlakshm) [reply src/sonic-config-engine/tests/test_cfggen.py]: fixed in latest commit

File: src/sonic-config-engine/tests/voq-sample-port-config.ini (hunk 3 above)
Comment (arlakshm) [reply src/sonic-config-engine/tests/voq-sample-port-config.ini]: fixed in latest commit

File: src/sonic-config-engine/minigraph.py (hunk 4 above)
Comment (arlakshm) [reply src/sonic-config-engine/minigraph.py:1537]: This is an internal port. There is no configuration for this port in minigraph, currently. We want to set this port to `admin-status` as `up`  always for everflow to work on Voq Chassis

**Review by ganglyu — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Review by ganglyu — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**